Library/ViewModels/AddNewCardViewModelTests.swift

```diff
@@ -177,64 +177,61 @@ internal final class AddNewCardViewModelTests: TestCase {
     self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.viewDidLoad()
+    self.vm.inputs.configure(with: .settings, project: nil)
 
     self.cardholderNameBecomeFirstResponder
       .assertValueCount(1, "Cardholder name field is first responder when view loads.")
     self.paymentDetailsBecomeFirstResponder.assertDidNotEmitValue("Not first responder when view loads")
     self.zipcodeTextFieldBecomeFirstResponder.assertDidNotEmitValue()
     self.vm.inputs.cardholderNameChanged("")
     self.vm.inputs.paymentInfo(isValid: false)
-    self.vm.inputs.cardBrand(isValid: false)
     self.vm.inputs.zipcodeChanged(zipcode: "")
-    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
+    self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.cardholderNameChanged("Native Squad")
     self.vm.inputs.cardholderNameTextFieldReturn()
     self.cardholderNameBecomeFirstResponder
       .assertValueCount(1, "Does not emit again.")
     self.paymentDetailsBecomeFirstResponder
       .assertValueCount(1, "First responder after editing cardholder name.")
-    self.saveButtonIsEnabled.assertValues([false], "Remains disabled while form is incomplete.")
+    self.saveButtonIsEnabled.assertDidNotEmitValue()
 
-    self.vm.inputs.creditCardChanged(cardDetails: ("4242 4242 4242 4242", 11, 99, "123"))
+    self.vm.inputs.creditCardChanged(cardDetails: ("4242 4242 4242 4242", 11, 99, "123", .visa))
+    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
     self.cardholderNameBecomeFirstResponder.assertValueCount(1, "Does not emit again.")
     self.paymentDetailsBecomeFirstResponder.assertValueCount(1, "Does not emit again.")
     self.vm.inputs.paymentInfo(isValid: true)
-    self.vm.inputs.cardBrand(isValid: true)
     self.vm.inputs.zipcodeChanged(zipcode: "123")
     self.saveButtonIsEnabled.assertValues([false, true], "Enabled when form is valid.")
   }
 
   func testSaveButtonEnabled() {
     self.saveButtonIsEnabled.assertDidNotEmitValue()
     self.vm.inputs.viewDidLoad()
+    self.vm.inputs.configure(with: .settings, project: nil)
 
     self.vm.inputs.cardholderNameChanged("")
     self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.paymentInfo(isValid: false)
-    self.vm.inputs.cardBrand(isValid: false)
     self.saveButtonIsEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.zipcodeChanged(zipcode: "")
-    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")
 
     self.vm.inputs.cardholderNameChanged("Native Squad")
-    self.vm.inputs.paymentInfo(isValid: true)
-    self.vm.inputs.cardBrand(isValid: true)
+    self.vm.inputs.creditCardChanged(cardDetails: ("6200 0000 0000 0005", 11, 99, "123", .unionPay))
```

Comment:
I noticed there's a couple of places in the tests where we call `paymentInfo(isValid)` after calling `creditCardChanged(cardDetails)`. We should try to keep our tests as close to what happens irl as we can - it looks like in the `STPPaymentCardTextFieldDelegate` we always call `creditCardChanges(cardDetails)` _after_ calling `paymentInfo(isValid:)`.

![image](https://user-images.githubusercontent.com/3156796/66329842-38deb480-e8fd-11e9-976d-32345b5d5364.png)

So this should be:

```
    self.vm.inputs.zipcodeChanged(zipcode: "")

    self.vm.inputs.cardholderNameChanged("Native Squad")
    self.vm.inputs.paymentInfo(isValid: false) // input called before .creditCardChanged(cardDetails:)
    self.vm.inputs.creditCardChanged(cardDetails: ("6200 0000 0000 0005", 11, 99, "123", .unionPay))
    self.saveButtonIsEnabled.assertValues([false], "Disabled form is incomplete")

    self.vm.inputs.zipcodeChanged(zipcode: "123") // missing field is entered
    self.vm.inputs.paymentInfo(isValid: true)
    self.vm.inputs.creditCardChanged(cardDetails: ("6200 0000 0000 0005", 11, 99, "123", .unionPay))

    self.saveButtonIsEnabled.assertValues([false, true], "Enabled when form is valid.")
```